client-server-client/src/main/scala/zio/raft/client/RaftClient.scala

```diff
@@ -0,0 +1,634 @@
+          } yield ()
+        
+        case rejection: SessionRejected =>
+          for {
+            _ <- ZIO.logWarning(s"Session rejected: ${rejection.reason}")
+            _ <- client.handleSessionRejected(rejection)
+            _ <- handleRejection(rejection, client)
+          } yield ()
+        
+        case other =>
+          ZIO.logWarning(s"Unexpected message in Connecting state: ${other.getClass.getSimpleName}")
+      }
+    
+    private def handleRejection(rejection: SessionRejected, client: RaftClient): UIO[Unit] = 
+      rejection.reason match {
+        case NotLeader =>
+          // Could implement leader redirection here
+          for {
+            _ <- ZIO.logInfo(s"Not leader, redirect to: ${rejection.leaderId}")
+            // Stay in Connecting, could retry with different leader
+          } yield ()
+          
+        case SessionNotFound =>
+          // Session doesn't exist, stay in Connecting and retry
+          ZIO.unit
+          
+        case _ =>
+          // Other errors - transition to Disconnected
+          client.transitionTo(Disconnected)
+      }
+  }
+  
+  /**
+   * Client has an active session and can send/receive requests.
+   * Contains all session-related data.
+   */
+  case class Connected(
+    sessionId: SessionId,
+    capabilities: Map[String, String],
+    createdAt: Instant,
+    serverRequestTracker: ServerRequestTracker
+  ) extends ClientState {
+    override def stateName: String = s"Connected($sessionId)"
+    
+    override def handleMessage(message: ServerMessage, client: RaftClient): UIO[Unit] = 
+      message match {
+        case response: ClientResponse =>
+          handleClientResponse(response, client)
+        
+        case KeepAliveResponse(timestamp) =>
+          ZIO.logDebug(s"Keep-alive acknowledged: $timestamp")
+        
+        case serverRequest: ServerRequest =>
+          handleServerRequest(serverRequest, client)
+        
+        case error: RequestError =>
+          handleRequestError(error, client)
+        
+        case SessionClosed(reason, leaderId) =>
+          for {
+            _ <- ZIO.logInfo(s"Session closed by server: $reason")
+            _ <- client.transitionTo(Disconnected)
+          } yield ()
+        
+        case other =>
+          ZIO.logWarning(s"Unexpected message in Connected state: ${other.getClass.getSimpleName}")
+      }
+    
+    private def handleClientResponse(response: ClientResponse, client: RaftClient): UIO[Unit] = 
+      for {
+        // Complete pending request via retry manager
+        _ <- client.retryManager.completePendingRequest(response.requestId, response.result)
+      } yield ()
+    
+    private def handleServerRequest(request: ServerRequest, client: RaftClient): UIO[Unit] = 
+      for {
+        _ <- ZIO.logInfo(s"Received server request: ${request.requestId}")
+        // User should call acknowledgeServerRequest after processing
```

Comment:
we should enqueue this and call acknowledgeServerRequest immediately. We don't need to wait, we just acknowledge the network operation here.